src/components/lib/c_stub.c

```diff
@@ -33,11 +33,11 @@ SS_ipc_client_fault(cos_flt_off flt)
 __attribute__((regparm(1))) int
 SS_ipc_client_marshal_args(struct usr_inv_cap *uc, long p0, long p1, long p2, long p3)
 {
-	return cos_sinv(uc->cap_no, 0, p0, p1, p2);
+	return cos_sinv(uc->cap_no, p0, p1, p2, p3);
 }
 
 __attribute__((regparm(1))) int
 SS_ipc_client_marshal_args_rets(struct usr_inv_cap *uc, long *r2, long *r3, long p0, long p1, long p2, long p3)
 {
-	return cos_sinv_rets(uc->cap_no, 0, p0, p1, p2, (word_t *)r2, (word_t *)r3);
+	return cos_sinv_rets(uc->cap_no, p0, p1, p2, p3, (word_t *)r2, (word_t *)r3);
```

Comment:
All of these changes to add the extra argument are great.  :+1:   Really good job.